### PR TITLE
Image viewer error

### PR DESCRIPTION
**Pull request by shaibv**
Added error state to image viewer
https://github.com/wix-private/wsr-issues/issues/63

### CODE REVIEW

File: src/ImageViewer/ImageViewer.js [hunk 1]
```diff
@@ -54,13 +57,33 @@ class ImageViewer extends WixComponent {
...
+          alignment="center"
+          textAlign="left"
+          content={this.props.errorMessage}
+          overlay=""
+          theme="dark"
+          maxWidth="230px"
+          hideDelay={150}
+          zIndex={10000}
+          >
+          <div data-hook="test" className={style.exclamation}><FormFieldError/></div>
```
Comment (lbelinsk) [inline src/ImageViewer/ImageViewer.js]: data-hook="test" ??

File: src/ImageViewer/ImageViewer.spec.js [hunk 2]
```diff
@@ -86,6 +86,26 @@ describe('ImageViewer', () => {
     });
   });
 
+
+  describe('Error state', () => {
+
+    it('should display error icon on error', () => {
```
Comment (lbelinsk) [inline src/ImageViewer/ImageViewer.spec.js]: missing test: "should not display by default"

File: src/ImageViewer/ImageViewer.js [hunk 3]
```diff
@@ -54,13 +57,33 @@ class ImageViewer extends WixComponent {
...
+        <Tooltip
+          dataHook="input-tooltip"
+          disabled={!this.props.errorMessage}
+          placement={this.props.tooltipPlacement}
+          alignment="center"
+          textAlign="left"
+          content={this.props.errorMessage}
+          overlay=""
+          theme="dark"
+          maxWidth="230px"
```
[the remark this answers is not in the document]
Comment (shaibv) [reply src/ImageViewer/ImageViewer.js]: 230 is the same as input error m should I change?
The width i see in the Zeplin is 222px so for sure it's not 204

[the remark this answers is not in the document]
Comment (lbelinsk) [reply src/ImageViewer/ImageViewer.js]: Go according to what the zeplin shows.

File: src/ImageViewer/ImageViewer.spec.js [hunk 4]
```diff
@@ -89,7 +88,19 @@ describe('ImageViewer', () => {
...
+        imageUrl: '',
+        width: 300,
+        height: 300
+      };
+
+      driver = createDriver(<ImageViewer {...props}/>);
+      expect(driver.isErrorVisible()).toBeFalsy();
+    });
+
+    it.only('should display error icon on error', () => {
```
Comment (lbelinsk) [inline src/ImageViewer/ImageViewer.spec.js]: only!!!!

File: src/ImageViewer/ImageViewer.spec.js [hunk 5]
```diff
@@ -101,6 +112,15 @@ describe('ImageViewer', () => {
 
       driver = createDriver(<ImageViewer {...props}/>);
       expect(driver.isErrorVisible()).toBeTruthy();
+      const wrapper = driver.getElement();
+      console.log('Element:', wrapper);
```
Comment (lbelinsk) [inline src/ImageViewer/ImageViewer.spec.js]: console.log!!

File: src/test-common.js [hunk 6]
```diff
@@ -58,3 +58,12 @@ export function flattenInternalDriver(driver) {
     return driver;
   }
 }
+
+
+export function resolveIn(timeout) {
```
Comment (lbelinsk) [inline src/test-common.js]: Good chance to point all other usages of this util func in other components.

File: src/ImageViewer/ImageViewer.spec.js (hunk 5 above)
Comment (shaibv) [reply src/ImageViewer/ImageViewer.spec.js]: it's still wip

File: src/ImageViewer/ImageViewer.spec.js [hunk 7]
```diff
@@ -86,6 +86,45 @@ describe('ImageViewer', () => {
...
+        imageUrl: '',
+        width: 300,
+        height: 300
+      };
+
+      driver = createDriver(<ImageViewer {...props}/>);
+      expect(driver.isErrorVisible()).toBeFalsy();
+    });
+
+    it('should display error icon on error', () => {
```
Comment (lbelinsk) [inline src/ImageViewer/ImageViewer.spec.js]: Missing a test to check that the tooltip disappears on mouse leave.
Moreover, this test `'should display error icon on error'` should test only that the error icon is visible, and you can do another test for the mouse leave and enter

Comment (shaibv) [reply src/ImageViewer/ImageViewer.spec.js]: This is responsibility of the tooltip component. 
It's shouldn't be tested in the imageViewer

File: src/ImageViewer/ImageViewer.driver.js [hunk 8]
```diff
@@ -11,11 +11,14 @@ const imageViewerDriverFactory = ({component, wrapper, element}) => {
...
   const updateImageButton = () => byHook('update-image');
   const removeImageButton = () => byHook('remove-image');
+  const errorIcon = () => byHook('error-tooltip');
 
   return {
+    getElement: () => element,
     getContainerStyles: () => element.getAttribute('style'),
     getImageUrl: () => image().getAttribute('src'),
     isImageVisible: () => !!image(),
+    isErrorVisible: () => !!errorIcon(),
```
[the remark this answers is not in the document]
Comment (shaibv) [reply src/ImageViewer/ImageViewer.driver.js]: Its does not have the functionality we need.

**Pull request merged**